tls timeout

Getting: [ERROR] [F9118AD1-991C-4C89-AD2F-7EAF93F3B555] [tls] timeout

Happening on cipher mismatch.
1. Test yourself with:  openssl s_client -connect  localhost:2525  -starttls smtp  -cipher ECDHE-ECDSA-AES128-SHA
2. wait for plugin.timeout time

Wondering if that's an error or not. And if it's okay, why is it logged with ERROR level then?

Did you wait enough time?  What's tls plugin timeout value?

```
 $ openssl s_client -connect  localhost:2525  -starttls smtp  -cipher ECDHE-ECDSA-AES128-SHA
CONNECTED(00000005)
140373926856360:error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure:s23_lib.c:184:

[DEBUG] [-] [core] plugin tls timeout is: 30s
...
[NOTICE] [D499022D-5F23-46A5-B1F8-91DE2E0A9F05] [core] disconnect ip=127.0.0.1 rdns="river.home" helo="openssl.client.net" relay=N early=N esmtp=Y tls=N pipe=N txns=0 rcpts=0/0/0 msgs=0/0/0 bytes=0 lr="" time=0.045
[ERROR] [D499022D-5F23-46A5-B1F8-91DE2E0A9F05] [tls] timeout
```

--- a/plugins/tls.js
+++ b/plugins/tls.js
@@ -96,6 +96,8 @@ exports.tls_unrecognized_command = function (next, connection, params) {
         return next(DENYSOFTDISCONNECT);
     }, timeout * 1000);
 
+    connection.notes.tls_timer = timer;
+
     /* Upgrade the connection to TLS. */
     connection.client.upgrade(plugin.tls_opts, function (authorized,
             verifyError, cert, cipher) {
@@ -122,3 +124,10 @@ exports.tls_unrecognized_command = function (next, connection, params) {
         });
     });
 };
+
+exports.hook_disconnect = function (next, connection) {
+    if (connection.notes.tls_timer) {
+        clearTimeout(connection.notes.tls_timer);
+    }
+    return next();
+}

